### PR TITLE
chore: add proxy health interval flag/env config

### PR DESCRIPTION
**Pull request by Emyrk**
Allows manually changing the proxy health interval

`--proxy-health-interval=10m`

### CODE REVIEW

File: codersdk/deployment.go [hunk 1]
```diff
@@ -1497,6 +1498,16 @@ Write out the current server config as YAML to stdout.`,
 			Default:     "", // empty string means pick best server
 			Hidden:      true,
 		},
+		{
+			Name:        "Proxy Health Check Interval",
+			Description: "The interval in which coderd should be checking the status of workspace proxies.",
+			Flag:        "proxy-health-interval",
+			Env:         "CODER_PROXY_HEALTH_INTERVAL",
+			Default:     (time.Minute).String(),
```
Comment (johnstcn) [inline codersdk/deployment.go:1506]:
```suggestion
			Default:     "1m",
```

Comment (Emyrk) [reply codersdk/deployment.go:1506]: https://github.com/coder/coder/blob/5ff22ac8e85fdbb2a8f00024d090eba337d267a7/codersdk/deployment.go#L535-L535

https://github.com/coder/coder/blob/5ff22ac8e85fdbb2a8f00024d090eba337d267a7/codersdk/deployment.go#L1087-L1087

We do this pattern for all other durations :shrug:

**Review by johnstcn — APPROVED** [review]
LGTM

File: codersdk/deployment.go (hunk 1 above)
Comment (johnstcn) [reply codersdk/deployment.go:1506]: Ah never mind then

File: cli/testdata/coder_server_--help.golden [hunk 2]
```diff
@@ -193,6 +193,10 @@ backed by Tailscale and WireGuard.
           The maximum lifetime duration users can specify when creating an API
           token.
 
+      --proxy-health-interval duration, $CODER_PROXY_HEALTH_INTERVAL (default: 1m0s)
```
Comment (kylecarbs) [inline cli/testdata/coder_server_--help.golden:196]: Should this be `health-check`?

Comment (Emyrk) [reply cli/testdata/coder_server_--help.golden:196]: Felt long to say `health check` and figured `health-interval` was descriptive enough

**Pull request merged**